src/redux/books/bookslice.js

```diff
@@ -1,26 +1,83 @@
-import { createSlice } from '@reduxjs/toolkit';
+/* eslint-disable no-param-reassign */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Mw6KlxQrMBAmuPW6nuKy/books';
+
+export const addBook = createAsyncThunk(
+  'books/addBook',
+  async (newBook) => {
+    const response = await axios.post(url, newBook);
+    return response.data;
+  },
+);
+
+export const getBooks = createAsyncThunk('books/getBooks', async () => {
+  const response = await axios.get(url);
+  const { data } = response;
+  const books = Object.keys(data).map((key) => ({
+    id: key,
+    ...data[key][0],
+  }));
+  return books;
+});
+
+export const removeBook = createAsyncThunk(
+  'books/removeBook',
+  async (id) => {
+    const response = await axios.delete(`${url}/${id}`);
+    return response.data;
+  },
```

Comment:
- [ ] The remove functionality does not work as expected as the book still displays on the screen even after the remove button is clicked. Please kindly fix this